### PR TITLE
Update selected item when overview items change

### PR DESCRIPTION
**Pull request by TheRealJon**
Update selected overview item when underlying resource changes to ensure that the change propagates up to the details panel.

### CODE REVIEW

File: frontend/public/components/overview.jsx [hunk 1]
```diff
@@ -367,6 +368,11 @@ class OverviewDetails extends React.Component {
       ...this.createStatefulSetItems()
     ];
 
+    // Ensure any changes to the selected item propagate back up to the side panel
+    if (!_.isEmpty(selectedItem)) {
+      selectItem(_.find(items, {obj: {metadata: {uid: selectedItem.obj.metadata.uid }}}));
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: style nit: extra space before closing `}`

Wonder if it's better to keep a map of UID -> obj for all objects and then just track the selected UID. Do we need to get object by UID anywhere else?

Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: Part of the complexity here is that the selected object needs to be somehow passed back up to the top level Overview component where the side panel is rendered. I've been thinking a lot about how this works, and there are probably other solutions, but I don't have any grand ideas. I don't think we select items by UID anywhere else. We find owned resources using UID, but I don't think a map would really help there.

Comment (spadgett) [reply frontend/public/components/overview.jsx]: It feels like this should be kept in redux rather than passing it up the component tree

**Comment by spadgett** [comment]
/lgtm

We can always come back to how we're handling selected item... Let's fix the bug. (Thanks.)

File: frontend/public/components/overview.jsx (hunk 1 above)
Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: Yeah, you're probably right. I could set up an overview redux object to manage state in this view. That would probably be a pretty big refactor though.

**Pull request merged**